app/controllers/listings/applications_controller.rb

```diff
@@ -13,6 +13,9 @@ def index
     end
 
     def new
+      if @listing.Lottery_Status != 'Not Yet Run'
+        redirect_to "/listings/#{@listing.Id}"
```

Comment:
you can probably use something more lexical like `listing_url` + param instead of interpolated string